[rush-lib] enable event hook scripts output by default

zecheng01 opened the pull request:
## Summary

The problem: currently, rush does not show any output from scripts defined within hooks like `postRushInstall ` by default. This is an issue because it prevents users from seeing any meaningful logs or progress from their scripts. This is more of a pain point when the scripts are long-running and may create the illusion of hanging.

The fix: we should enable event hook scripts outputs by default, since they are fully controlled by users, and they shouldn't be hidden behind `verbose` or `debug` flag, as it would create unwanted noise from other steps when enabled and hinder the flexibility of user extension points. 

The only change is to let user script to handle output for event hooks.

Fixes #553

## How it was tested

- test default experiments.json file
  - `rush init`
  - verify common/config/rush/experiments.json has `printEventHooksOutputToConsole` flag
- configure event hooks such as 'postRushInstall' with inline command and shell script invocation
- test default behaviour
  - run `rush update`
  - expect no output from event hooks
- test debug feature
  - run `rush --debug update`
  - expect outputs from event hooks printed to console
- test experiment feature: 
  - enable `printEventHooksOutputToConsole` flag in common/config/rush/experiments.json
  - run `rush update`
  - expect outputs from event hooks printed to console

zecheng01 commented:
> @zecheng01 please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Microsoft"

iclanton commented:
I think we should put this behind an experiment. @zecheng01 - Would you mind making adding that?

zecheng01 commented:
> I think we should put this behind an experiment. @zecheng01 - Would you mind making adding that?

@iclanton Thank you for the suggestion, I have added `printEventHooksOutputToConsole` experiment flag, updated the manual test cases, and the new behaviour is now off by default.

iclanton commented:
This has been released with Rush 5.95.0
